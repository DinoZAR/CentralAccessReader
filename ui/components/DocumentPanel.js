import React from 'react';

import { StyleSheet, css } from '../styles/aphrodite';
import constants from '../styles/constants';

class DocumentPanel extends React.Component {
  render() {
    return (
      <div className={css(styles.window)}>
        <div className={css(styles.panel)}>
          <p>Some cool text to have here</p>
        </div>
      </div>
    );
  }
}

const styles = StyleSheet.create({
  window: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
  },
  panel: {
    backgroundColor: 'black',
    color: 'white',
    maxWidth: '50rem',
    width: '100%',
    padding: '1rem',
    boxShadow: constants.panelShadow,
  }
});

export default DocumentPanel;
