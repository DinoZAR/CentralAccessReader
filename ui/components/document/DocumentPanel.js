import React from 'react';
import PropTypes from 'prop-types';

import { StyleSheet, css } from '../../styles/aphrodite';
import constants from '../../styles/constants';

import DocumentContent from './DocumentContent';

class DocumentPanel extends React.Component {
  render() {
    const { content } = this.props;
    return (
      <div className={css(styles.window)}>
        <div className={css(styles.panel)} style={{ maxWidth: '50rem' }}>
          {content.map(c => <DocumentContent key={c.id} content={c} />)}
        </div>
      </div>
    );
  }
}

DocumentPanel.propTypes = {
  content: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
  })).isRequired,
};

const styles = StyleSheet.create({
  window: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
  },
  panel: {
    backgroundColor: 'black',
    color: 'white',
    width: '100%',
    padding: '2rem',
    boxShadow: constants.panelShadow,
  }
});

export default DocumentPanel;
