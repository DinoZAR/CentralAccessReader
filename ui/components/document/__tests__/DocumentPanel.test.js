import React from 'react';
import DocumentPanel from '../DocumentPanel';

describe('DocumentPanel', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DocumentPanel content={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
