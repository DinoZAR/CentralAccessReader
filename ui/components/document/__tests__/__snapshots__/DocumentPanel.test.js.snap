// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentPanel renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "maxWidth": "50rem",
      }
    }
  />
</div>
`;
